### PR TITLE
Use latest knowledge context tool

### PR DESCRIPTION
**Pull request by StrongMonkey**
This PR use the latest and greated context tool from @iwilltry42. It also removes the inline tool and replace it with the remote tool that is capable of querying data from both thread and script dataset.

### CODE REVIEW

File: server/app.mjs [hunk 1]
```diff
@@ -327,17 +330,6 @@ const mount = async (
     for (let block of script) {
       if (block.type !== 'text') {
         block.name = newName || block.name;
-        block.tools = [
```
Comment (StrongMonkey) [inline server/app.mjs:330]: @thedadams We actually change it back to be a remote tool so it is easy to work with. So no longer need to do special logic when copying now. Sorry about that.

File: components/chat.tsx [hunk 2]
```diff
@@ -93,6 +94,7 @@ const Chat: React.FC<ScriptProps> = ({
         formValues,
         workspace,
         thread,
+        scriptId,
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply components/chat.tsx:97]: Yes... I will remove it.

[the remark this answers is not in the document]
Comment (StrongMonkey) [reply components/chat.tsx]: Removed

**Review by iwilltry42 — APPROVED** [review]

**Review by thedadams — APPROVED** [review]

File: components/scripts/script-save.tsx [hunk 3]
```diff
@@ -95,28 +87,16 @@ const SaveScriptDropdown = () => {
...
       await clearThreadKnowledge(workspace);
-
-      if (
-        !scriptContent.find(
-          (t) => t.type !== 'text' && t.name === KNOWLEDGE_NAME
-        )
-      ) {
-        newKnowledgeToolBlock = await assistantKnowledgeTool(scriptID, 10);
-      }
+      return;
```
Comment (thedadams) [inline components/scripts/script-save.tsx:90]: nit: not needed

**Pull request merged**